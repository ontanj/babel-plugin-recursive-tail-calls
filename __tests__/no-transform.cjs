const babel = require('@babel/core');
const plugin = require("..");

const noRecursion = `
function noRecursion(a) {
  return func(a)
}
`;

const notTail = `
function notTail(a) {
  const c = notTail(a);
  return c;
}
`;

const cases = Object.entries({
  noRecursion,
  notTail
});

describe("no-transform", () => {
  it.each(cases)('%s', (_, testCase) => {
    const { code } = babel.transform(testCase, { plugins: [plugin] });
    expect(code).toMatchSnapshot();
  });
});
