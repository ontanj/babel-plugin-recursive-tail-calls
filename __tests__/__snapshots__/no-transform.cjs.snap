// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no-transform noRecursion 1`] = `
"function noRecursion(a) {
  return func(a);
}"
`;

exports[`no-transform notTail 1`] = `
"function notTail(a) {
  const c = notTail(a);
  return c;
}"
`;
